test/e2e/fixtures.go

```diff
@@ -130,42 +129,44 @@ func setupTest(tb testing.TB) (*TestData, error) {
 }
 
 func setupTestWithIPFIXCollector(tb testing.TB) (*TestData, error, bool) {
-	data := &TestData{}
+	// TODO: remove hardcoding to IPv4 after flow aggregator supports IPv6
 	isIPv6 := false
-	if err := data.setupLogDirectoryForTest(tb.Name()); err != nil {
-		tb.Errorf("Error creating logs directory '%s': %v", data.logsDirForTestCase, err)
+	if err := testData.setupLogDirectoryForTest(tb.Name()); err != nil {
+		tb.Errorf("Error creating logs directory '%s': %v", testData.logsDirForTestCase, err)
 		return nil, err, isIPv6
 	}
-	tb.Logf("Creating K8s clientset")
-	if err := data.createClient(); err != nil {
+	tb.Logf("Creating '%s' K8s Namespace", testNamespace)
+	if err := ensureAntreaRunning(tb, testData); err != nil {
 		return nil, err, isIPv6
 	}
-	tb.Logf("Creating '%s' K8s Namespace", testNamespace)
-	if err := data.createTestNamespace(); err != nil {
+	if err := testData.createTestNamespace(); err != nil {
 		return nil, err, isIPv6
 	}
 	// Create pod using ipfix collector image
-	if err := data.createPodOnNode("ipfix-collector", "", ipfixCollectorImage, nil, nil, nil, nil, true, nil); err != nil {
+	if err := testData.createPodOnNode("ipfix-collector", "", ipfixCollectorImage, nil, nil, nil, nil, true, nil); err != nil {
 		tb.Errorf("Error when creating the ipfix collector Pod: %v", err)
 	}
-	ipfixCollectorIP, err := data.podWaitForIPs(defaultTimeout, "ipfix-collector", testNamespace)
+	ipfixCollectorIP, err := testData.podWaitForIPs(defaultTimeout, "ipfix-collector", testNamespace)
 	if err != nil || len(ipfixCollectorIP.ipStrings) == 0 {
 		tb.Errorf("Error when waiting to get ipfix collector Pod IP: %v", err)
+		return nil, err, isIPv6
 	}
-	tb.Logf("Applying Antrea YAML with ipfix collector address")
-	ipStr := ipfixCollectorIP.ipStrings[0]
-	if net.ParseIP(ipStr).To4() == nil {
-		ipStr = fmt.Sprintf("[%s]", ipStr)
-		isIPv6 = true
+	ipStr := ipfixCollectorIP.ipv4.String()
+
+	tb.Logf("Applying flow aggregator YAML with ipfix collector address: %s:%s:tcp", ipStr, ipfixCollectorPort)
+	faClusterIP, err := testData.deployFlowAggregator(fmt.Sprintf("%s:%s:tcp", ipStr, ipfixCollectorPort))
+	if err != nil {
+		return testData, err, isIPv6
 	}
-	if err := data.deployAntreaFlowExporter(fmt.Sprintf("%s:%s:tcp", ipStr, ipfixCollectorPort)); err != nil {
-		return data, err, isIPv6
+	tb.Logf("Deploying flow exporter with collector address: %s:%s:tcp", faClusterIP, ipfixCollectorPort)
+	if err = testData.deployAntreaFlowExporter(fmt.Sprintf("%s:%s:tcp", faClusterIP, ipfixCollectorPort)); err != nil {
```

Comment:
I believe we should only use the ClusterIP on Kind (since apparently there are DNS issue, which should be explained in a comment). On Jenkins we should use the default (DNS path), since this is what users will use by default and it's better to test for it.

Reply:
Yes. Thought of doing this later once everything runs perfectly on jenkins. Added it now.